### PR TITLE
Stop expunging instances at TaskKiller level

### PR DESCRIPTION
**Pull request by meln1k**
Summary: expunging is moved to KillService actor, it use a newly introduced `wipe` parameter to correctly expunge instances.

JIRA issues: https://jira.mesosphere.com/browse/MARATHON-8531

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala [hunk 1]
```diff
@@ -424,8 +424,13 @@ class SchedulerActions(
               else instanceTracker.setGoal(i.instanceId, Goal.Decommissioned)
             }
 
+            logger.info("waiting for the changeGoalsFuture")
```
Comment (jeschkies) [inline src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala]: This seems like a debug log.

File: src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala [hunk 2]
```diff
@@ -424,8 +424,13 @@ class SchedulerActions(
...
             await(Future.sequence(changeGoalsFuture))
-            await(killService.killInstances(instances, KillReason.OverCapacity))
+            logger.info(s"waiting for the killService.killInstances(${instances.map(i => i.instanceId -> i.state)}, KillReason.OverCapacity)")
```
Comment (jeschkies) [inline src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala]: Should we log this as debug?

File: src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala [hunk 3]
```diff
@@ -424,8 +424,13 @@ class SchedulerActions(
...
+
+            val res = await(killService.killInstances(instances, KillReason.OverCapacity, wipe = false))
+            logger.info(s"waiting for the killService.killInstances - done")
```
Comment (jeschkies) [inline src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala]: Same here.

File: src/main/scala/mesosphere/marathon/core/task/termination/impl/KillAction.scala [hunk 4]
```diff
@@ -75,8 +75,13 @@ private[termination] object KillAction extends StrictLogging {
       val msg = if (isUnkillable) s"its condition is ${maybeCondition.fold("unknown")(_.toString)}"
       else "none of its tasks are running"
       if (hasReservations) {
-        logger.info(s"Ignoring kill request for $instanceId; killing it while $msg is unsupported")
-        KillAction.Noop
+        if (wipe) {
+          logger.warn(s"Expunging $instanceId from state because wipe = true")
+          KillAction.ExpungeFromState
+        } else {
+          logger.info(s"Ignoring kill request for $instanceId; killing it while $msg is unsupported")
```
Comment (jeschkies) [inline src/main/scala/mesosphere/marathon/core/task/termination/impl/KillAction.scala]: What do you mean by "killing it while"?

File: src/test/scala/mesosphere/marathon/core/task/termination/impl/KillServiceActorTest.scala [hunk 5]
```diff
@@ -340,6 +372,15 @@ class KillServiceActorTest extends AkkaUnitTest with StrictLogging {
       TestInstanceBuilder.newBuilder(appId).addTaskWithBuilder().taskForStatus(mesosState, stagedAt).build().getInstance()
     }
 
+    def mockReservedInstance(appId: PathId, stagedAt: Timestamp, mesosState: mesos.Protos.TaskState): Instance = {
```
Comment (alenkacz) [inline src/test/scala/mesosphere/marathon/core/task/termination/impl/KillServiceActorTest.scala]: I would say it is not a mock, it is real instance. Do we really need this helper anyway?

File: src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala (hunk 1 above)
Comment (meln1k) [reply src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala]: Oh, yeah, forgot about that! Good catch!

**Pull request closed without merging**